Documentation/ceph-backup.md

```diff
@@ -0,0 +1,483 @@
+
+```console
+kubectl get svc -n rook-ceph rook-ceph-mon-a -o yaml > mon-service-a.yaml
+kubectl get svc -n rook-ceph rook-ceph-mon-b -o yaml > mon-service-b.yaml
+kubectl get svc -n rook-ceph rook-ceph-mon-c -o yaml > mon-service-c.yaml
+```
+
+For each of those files we have to delete `metadata.creationTimestamp`, `metadata.ownerReferences`, `metadata.resourceVersion`, `metadata.selfLink` and `metadata.uid`.
+
+For example for file `mon-service-a.yaml`:
+
+```yaml
+apiVersion: v1
+kind: Service
+metadata:
+  creationTimestamp: "2020-10-04T19:22:31Z"
+  labels:
+    app: rook-ceph-mon
+    ceph_daemon_id: a
+    ceph_daemon_type: mon
+    mon: a
+    mon_cluster: rook-ceph
+    pvc_name: rook-ceph-mon-a
+    rook_cluster: rook-ceph
+  name: rook-ceph-mon-a
+  namespace: rook-ceph
+  ownerReferences:
+  - apiVersion: ceph.rook.io/v1
+    blockOwnerDeletion: true
+    controller: true
+    kind: CephCluster
+    name: rook-ceph
+    uid: 7561619e-bd35-4649-8af7-a7a0dd7c11b4
+  resourceVersion: "3046"
+  selfLink: /api/v1/namespaces/rook-ceph/services/rook-ceph-mon-a
+  uid: 77e161de-d9ed-41eb-b9c2-386393d21369
+spec:
+  clusterIP: 10.4.1.210
+  ports:
+  - name: tcp-msgr1
+    port: 6789
+    protocol: TCP
+    targetPort: 6789
+  - name: tcp-msgr2
+    port: 3300
+    protocol: TCP
+    targetPort: 3300
+  selector:
+    app: rook-ceph-mon
+    ceph_daemon_id: a
+    mon: a
+    mon_cluster: rook-ceph
+    rook_cluster: rook-ceph
+  sessionAffinity: None
+  type: ClusterIP
+status:
+  loadBalancer: {}
+```
+
+We end up with:
+```yaml
+apiVersion: v1
+kind: Service
+metadata:
+  labels:
+    app: rook-ceph-mon
+    ceph_daemon_id: a
+    ceph_daemon_type: mon
+    mon: a
+    mon_cluster: rook-ceph
+    pvc_name: rook-ceph-mon-a
+    rook_cluster: rook-ceph
+  name: rook-ceph-mon-a
+  namespace: rook-ceph
+spec:
+  clusterIP: 10.4.1.210
+  ports:
+  - name: tcp-msgr1
+    port: 6789
+    protocol: TCP
+    targetPort: 6789
+  - name: tcp-msgr2
+    port: 3300
+    protocol: TCP
+    targetPort: 3300
+  selector:
+    app: rook-ceph-mon
+    ceph_daemon_id: a
+    mon: a
+    mon_cluster: rook-ceph
+    rook_cluster: rook-ceph
+  sessionAffinity: None
+  type: ClusterIP
+status:
+  loadBalancer: {}
+```
+
+## Restore Back Up
+
+Execute the following command to restore the backup.
+
+```console
+velero restore create --from-backup <BackupName>
+```
+
+Delete the `MON` services backed up by Velero and restore the old ones.
+
+```console
+# Delete MONs
+kubectl delete svc -n rook-ceph rook-ceph-mon-a
+kubectl delete svc -n rook-ceph rook-ceph-mon-b
+kubectl delete svc -n rook-ceph rook-ceph-mon-c
+
+# Restore the new ones
+kubectl apply -f mon-service-a.yaml
+kubectl apply -f mon-service-b.yaml
+kubectl apply -f mon-service-c.yaml
+```
+
+After that we need to create cluster.
+
+```console
+kubectl apply -f common.yaml
+kubectl apply -f operator.yaml
+kubectl apply -f cluster-on-pvc.yaml
+```
+
+Owner reference will be attached to `MON` services.
```

Comment:
Why mention the owner reference?

Reply:
When we export the service for the first time we get a yaml as follows:

```yaml
apiVersion: v1
kind: Service
metadata:
  creationTimestamp: "2020-10-04T19:22:31Z"
  labels:
    app: rook-ceph-mon
    ceph_daemon_id: a
    ceph_daemon_type: mon
    mon: a
    mon_cluster: rook-ceph
    pvc_name: rook-ceph-mon-a
    rook_cluster: rook-ceph
  name: rook-ceph-mon-a
  namespace: rook-ceph
  ownerReferences:
  - apiVersion: ceph.rook.io/v1
    blockOwnerDeletion: true
    controller: true
    kind: CephCluster
    name: rook-ceph
    uid: 7561619e-bd35-4649-8af7-a7a0dd7c11b4
  resourceVersion: "3046"
  selfLink: /api/v1/namespaces/rook-ceph/services/rook-ceph-mon-a
  uid: 77e161de-d9ed-41eb-b9c2-386393d21369
spec:
  clusterIP: 10.4.1.210
  ports:
  - name: tcp-msgr1
    port: 6789
    protocol: TCP
    targetPort: 6789
  - name: tcp-msgr2
    port: 3300
    protocol: TCP
    targetPort: 3300
  selector:
    app: rook-ceph-mon
    ceph_daemon_id: a
    mon: a
    mon_cluster: rook-ceph
    rook_cluster: rook-ceph
  sessionAffinity: None
  type: ClusterIP
status:
  loadBalancer: {}
```

I am explaining that you need to delete `metadata.ownerReferences`. However, they will get added by the operator in the recreation